src/ellipse2d.jl

```diff
@@ -1,18 +1,18 @@
 #Ellipse methods
 
-Ellipse(x::Int, y::Int, ρx::T, ρy::U , thickness::Int ,fill::Bool) where {T<:Real, U<:Real} = Ellipse(Point(x,y), ρx, ρy, thickness,fill)
-Ellipse(p::CartesianIndex{2}, ρx::T, ρy::U, thickness::Int ,fill::Bool) where {T<:Real, U<:Real} = Ellipse(Point(p), ρx, ρy,thickness,fill)
-Ellipse(circle::CirclePointRadius) = Ellipse(circle.center, circle.ρ, circle.ρ,circle.thickness,circle.fill)
+Ellipse(x::Int, y::Int, ρx::T, ρy::U; thickness::Int =  0, fill::Bool = true) where {T<:Real, U<:Real} = Ellipse(Point(x,y), ρx, ρy, thickness, fill)
+Ellipse(p::CartesianIndex{2}, ρx::T, ρy::U; thickness::Int = 0, fill::Bool = true) where {T<:Real, U<:Real} = Ellipse(Point(p), ρx, ρy, thickness, fill)
+Ellipse(circle::CirclePointRadius) = Ellipse(circle.center, circle.ρ, circle.ρ, circle.thickness, circle.fill)
 
 function draw!(img::AbstractArray{T, 2}, ellipse::Ellipse, color::T) where T<:Colorant
 	ys = Int[]
 	xs = Int[]
 	break_point = 0
     if ellipse.thickness >= ellipse.ρy || ellipse.thickness >= ellipse.ρx
-			error("Thickness greater than a,b not allowed")
+		error("Thickness greater than a,b not allowed")
 	end
```

Comment:
We need to early raise errors whenever we can, so the "correct" place to do these argument checks are in the inner constructor:

```julia
struct Ellipse{T<:Real, U<:Real} <: Drawable
    center::Point
    ρx::T
    ρy::U
    thickness::Int
    fill::Bool
    function Ellipse{T, U}(center, ρx::T, ρy::U, thickness::Int, fill::Bool) where {T<:Real, U<:Real}
        thickness >= min(ρx, ρy) && throw(ArgumentError("Thickness $thickness should be smaller than $(min(ρx, ρy))."))
        new{T, U}(center, ρx, ρy, thickness, fill)
    end
end
```